---
layout: archive
title: "People"
permalink: /people/
author_profile: true
toc: true
---

{% include base_path %}

CoBiODA is the Bioinformatics Hub of the Institute of Molecular and Cellular Pharmacology of CNRS located in Valbonne Sophia-Antipolis, Nice, France. It aims at setup a bioinformatics platform dedicated to the data analysis support of the IPMC research teams allowing an emulation of the bioinformatics resources within a common structure.


<table cellspacing="0" cellpadding="0" border="0">
<tr>
  <td width="20%"><img src="/images/diamant.jpg" alt="diamant" title="diamant" width="150" align="left"/></td>
  <td  width="80%" valign="top">PhD UniCA in Bernard Mari and Georges Vassaux team, I'm working on development of computational method for the anaysis and studying of isoform-level expression at he single cell resolution using long-read sequencing</td>
</tr>
</table>

<table cellspacing="0" cellpadding="0" border="0">
<tr>
  <td width="20%"><img src="/images/mcandrew.jpg" alt="mcandrew" title="mcandrew" width="200" /></td>
  <td width="80%" valign="top">PhD LongTREC consortium in Pascal Barbry team. Initially specialized in wet-lab and engineering, with emphasis on industrial drug production, i transitioned to a computational focus during my Masters program at University of Galway. I'm a PhD candidate in Pascal Barbry's team on the subject "Long-read Single-Cell and Spatial Transcriptomics to Uncover the Diversity of Isoform Expression". I aim to develop an Isoform Resolution Single-cell Atlas with a robust and scalable web application, to develop Artificial Intelligence tools to refine the assessment and interpretation of isoform differences and hypothesise isoform functionalities and apply spatial transcriptomics approaches to analyse the spatial patterning of splicing factor expression and determine its role in isoform diversity to decipher alternativeÂ splicing regulatory mechanisms.</td>
</tr>
</table>

<table cellspacing="0" cellpadding="0" border="0">
<tr>
  <td width="20%"><img src="/images/test.jpg" alt="truchi" title="truchi" width="200" /></td>
  <td width="80%" valign="top">CDD UniCA</td>
</tr>
</table>

<table cellspacing="0" cellpadding="0" border="0">
<tr>
  <td width="20%"><img src="/images/benkhedher.jpg" alt="benkhedher" title="benkhedher" width="150" /></td>
  <td width="80%" valign="top">CDD in Bernard Mari and Georges Vassaux team</td>
</tr>
</table>

<table cellspacing="0" cellpadding="0" border="0">
<tr>
  <td border="0" width="20%"><img src="/images/lebrigand.jpg" alt="lebrigand" title="lebrigand" width="200" /></td>
  <td border="0" width="80%" valign="top">Head of Bioinformatics at the Institute of Molecular and Cellular Pharmacology of CNRS in Nice-Sophia-Antipolis. Since 2015 i focused my expertise on methodological developments around single cell, and more recently on spatial transcriptomics, using 10x Genomics Chromium and Visium libraries sequenced on a Nanopore PromethION long-read sequencer to perform isoform-level profiling at the spatial resolution of the single cell. I'm an expert of R and Python statistical analysis of genomics data, focusing my interest on imaging-based spatial transcriptomics data analysis.</td>
</tr>
</table>
