---
title: "4th Workshop Single-Cell and Spatial multiomics"
collection: teaching
type: "Graduate course"
permalink: /teaching/sincellTE22
venue: "Roscoff"
date: 2022-01-01
location: "Roscoff, FR"
excerpt: ''
---

[Access Program](https://moodle.france-bioinformatique.fr/course/view.php?id=7)


![4th Workshop Single-Cell and Spatial multiomics, Roscoff 2022](/images/sincellTE22.png "4th Workshop Single-Cell and Spatial multiomics, Roscoff 2022")

Duration
======

This fourth edition of the school will be held from 09 January 2022 to 14th january 2022 at the Roscoff Biological Station (France).

Objectives
======

This workshop focuses on the large-scale study of heterogeneity across individual cells from a genomic, transcriptomic and epigenomic point of view. New technological developments enable the characterization of molecular information at a single cell resolution for large numbers of cells. The high dimensional omics data that these technologies produce raise novel methodological challenges for the analysis. In this regard, dedicated bioinformatics and statistical methods have been developed in order to extract robust information.

The workshop aims to provide such methods for engineers and researchers directly involved in functional genomics projects making use of single-cell technologies. A wide range of single cell topics will be covered in lectures, demonstrations and practical classes. Among others, the areas and issues to be addressed will include the choice of the most appropriate single-cell sequencing technology, the experimental design and the bioinformatics and statistical methods and pipelines. For this edition, new courses/practicals will focus on spatial transcriptomics, cell phenotyping and additional multi-omics.

Participants
======

This course is directed towards 30 engineers and researchers who regularly need to undertake single-cell data analysis as well as PhD candidates and Postdocs in computational biology or bioinformatics that are interested in the development of methods and pipelines for high dimension single-cell data analysis.

Working environment
======

Unix command will be used for bioinformatic analysis, and R programming language for statistical analysis.

Prerequisites
======

Participants must have prior experience on NGS data analysis  with everyday use of R and good knowledge of Unix command line. Before the training, participants will be asked to familiarize themselves with the processing and primary analyses steps of scRNA-seq datasets with provided pedagogic material. Study material : Already working on his/her own single cell dataset is not mandatory.
