---
title: "The spatial landscape of gene expression isoforms in tissue sections"
collection: publications
permalink: /publication/2023-SiT
excerpt: ''
date: 2023-05-08
venue: 'Nucleic Acids Research'
paperurl: 'https://academic.oup.com/nar/article/51/8/e47/7079641'
citation: ''
---

In situ capturing technologies add tissue context to gene expression data, with the potential of providing a greater understanding of complex biological systems. However, splicing variants and full-length sequence heterogeneity cannot be characterized at spatial resolution with current transcriptome profiling methods. To that end, we introduce spatial isoform transcriptomics (SiT), an explorative method for characterizing spatial isoform variation and sequence heterogeneity using long-read sequencing. We show in mouse brain how SiT can be used to profile isoform expression and sequence heterogeneity in different areas of the tissue. SiT reveals regional isoform switching of Plp1 gene between different layers of the olfactory bulb, and the use of external single-cell data allows the nomination of cell types expressing each isoform. Furthermore, SiT identifies differential isoform usage for several major genes implicated in brain function (Snap25, Bin1, Gnas) that are independently validated by in situ sequencing. SiT also provides for the first time an in-depth A-to-I RNA editing map of the adult mouse brain. Data exploration can be performed through an online resource ([https://www.isomics.eu](https://www.isomics.eu)), where isoform expression and RNA editing can be visualized in a spatial context.


Lebrigand K. et al., "The spatial landscape of gene expression isoforms in tissue sections" <i>Nucleic Acids Research</i> (2023).
